src/apps/main/antivirus/utils/dbConections.ts

```
@@ -0,0 +1,48 @@
+import { DatabaseCollectionAdapter } from '../../database/adapters/base';
+import { ScannedItem } from '../../database/entities/FileSystemHashed';
+
+export class DBScannerConnection {
+  constructor(private db: DatabaseCollectionAdapter<ScannedItem>) {}
+
+  addItemToDatabase = async (item: ScannedItem): Promise<boolean> => {
+    const currentTime = new Date().toISOString();
+
+    const itemToAdd: ScannedItem = {
+      ...item,
+      createdAt: currentTime,
+      updatedAt: currentTime,
+    };
+
+    try {
+      const createdItem = await this.db.create(itemToAdd);
+      return createdItem.success;
+    } catch (error) {
+      return false;
```

Error is being lost here